### PR TITLE
Split API and CLI modules

### PR DESCRIPTION
**Pull request by aalmiray**
Fix for #29

### CODE REVIEW

File: pom.xml
Comment (gunnarmorling) [inline pom.xml:41]: How about renaming "launcher" to "core" and "cli" to "launcher"? In the currently used terminology, "launcher" refers to the ready-made main class (as opposed to the API), so that'd be consistent.

File: vertx-example/layrry-links-runner/src/test/java/com/example/layrry/links/integrationtest/LayrryLinksTest.java
Comment (gunnarmorling) [inline vertx-example/layrry-links-runner/src/test/java/com/example/layrry/links/integrationtest/LayrryLinksTest.java:61]: Following to my comment above, how about `LayrryLauncher`?

File: layrry-cli/src/main/java/org/moditect/layrry/cli/LayrryRunner.java
Comment (gunnarmorling) [inline layrry-cli/src/main/java/org/moditect/layrry/cli/LayrryRunner.java:37]: Any strong reason for not doing this in `main()`?

File: layrry-cli/src/main/java/org/moditect/layrry/cli/LayrryRunner.java
Comment (gunnarmorling) [inline layrry-cli/src/main/java/org/moditect/layrry/cli/LayrryRunner.java:28]: s/Layrrry/LayrryLauncher/

File: layrry-launcher/src/main/java/org/moditect/layrry/Layrry.java
Comment (gunnarmorling) [inline layrry-launcher/src/main/java/org/moditect/layrry/Layrry.java:33]: Should use `Path`.

That said, I'm a bit confused :) If we still have the reference to config parsers in this module, what do we actually gain by this split?

**Review by gunnarmorling — COMMENTED** [review]
Thanks, @aalmiray. A few comments/questions inline. Could you hold off with all this though a bit until #33 is merged, as I think this will create some merge conflicts. Thanks!

File: layrry-cli/src/main/java/org/moditect/layrry/cli/LayrryRunner.java
Comment (aalmiray) [reply layrry-cli/src/main/java/org/moditect/layrry/cli/LayrryRunner.java:37]: Not really, no. It just feels "odd" to invoke `main` directly.

File: layrry-launcher/src/main/java/org/moditect/layrry/Layrry.java
Comment (aalmiray) [reply layrry-launcher/src/main/java/org/moditect/layrry/Layrry.java:33]: The `layrry-core` module defines the core API but does not depend on an specific config format. This lets developers embed Layrry in their custom launchers and pick/force a config format if they want to.

The `layrry-launcher` provides a ready made launcher with predefined config formats, in a single fat JAR.

There are 2 ways to launch a Layrry powered app:
1. provided a configuration file that bootstraps the `Layers` instance to use.
2. provide a programmatically configured `Layers` instance.

The `Layrry` class takes care of 1) while `Layers` takes care of 2) as the `Layers` class already exposes a method for running the computed layers. But, if `Layrry` is to become the single entry point then it should have a method such as

    void run(Layers layers, String main, String... args)

**Comment by aalmiray** [comment]
Be advised that Layrry's jbang-catalog will have to be updated once this change is merged.

**Review by gunnarmorling — APPROVED** [review]

**Comment by gunnarmorling** [comment]
Merged; thx, @aalmiray!

**Pull request merged**
